cmd/inspect.go

```diff
@@ -197,19 +209,97 @@ func printContainerInspect(c *clab.CLab, containers []types.GenericContainer, fo
 		return nil
 	}
 
-	runtime := c.GlobalRuntime()
-
-	stdout, stderr, err := runtime.Exec(context.Background(), mysocketCID, []string{"mysocketctl", "socket", "ls"})
+	// search for the mysocketio token file by deducing it from the topology config binds section
+	tokenFile, err := deduceMySocketIoTokenFileFromBindMounts(c.Nodes, c.TopoFile.GetDir())
 	if err != nil {
-		return fmt.Errorf("failed to execute cmd: %v", err)
-
+		return err
 	}
-	if len(stderr) > 0 {
-		log.Infof("errors during listing mysocketio sockets: %s", string(stderr))
+	// retrieve the MySocketIO Data
+	socketdata, err := getMySocketIoData(tokenFile)
+	if err != nil {
+		return fmt.Errorf("error when processing mysocketio data: %v", err)
 	}
 
+	// prepare data for table
+	var tabDataMySocketIo [][]string
+	for _, entry := range socketdata {
+		var portstrarr []string
+		for _, port := range entry.Ports {
+			portstrarr = append(portstrarr, strconv.Itoa(port))
+		}
+		tabDataMySocketIo = append(tabDataMySocketIo, []string{*entry.SocketId, *entry.DnsName, strings.Join(portstrarr, ", "), *entry.Type, strconv.FormatBool(entry.CloudAuth), *entry.Name})
+	}
+	tableMySocketIo := tablewriter.NewWriter(os.Stdout)
+	headerMySocketIo := []string{
+		"Socket ID",
+		"DNS Name",
+		"Ports",
+		"Type",
+		"Cloud Auth",
+		"Name",
+	}
+	// configure table output
+	tableMySocketIo.SetHeader(headerMySocketIo)
+	tableMySocketIo.SetAutoFormatHeaders(false)
+	tableMySocketIo.SetAutoWrapText(false)
+	tableMySocketIo.AppendBulk(tabDataMySocketIo)
 	fmt.Println("Published ports:")
-	fmt.Println(string(stdout))
+	tableMySocketIo.Render()
 
 	return nil
 }
+
+// getMySocketioData uses the mysocketio.http client to retrieve the socket data
+func getMySocketIoData(tokenfile string) ([]*types.MySocketIoEntry, error) {
+	result := []*types.MySocketIoEntry{}
+
+	client, err := mysocketio.NewClient(tokenfile)
+	if err != nil {
+		return nil, err
+	}
+
+	sockets := []mysocketio.Socket{}
+	err = client.Request("GET", "connect", &sockets, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range sockets {
+		newentry := &types.MySocketIoEntry{
+			SocketId:  &sockets[i].SocketID,
+			DnsName:   &sockets[i].Dnsname,
+			Ports:     sockets[i].SocketTcpPorts,
+			Type:      &sockets[i].SocketType,
+			CloudAuth: sockets[i].CloudAuthEnabled,
+			Name:      &sockets[i].Name,
+		}
+		result = append(result, newentry)
+	}
+	return result, nil
+}
+
+// deduceMySocketIoTokenFileFromBindMounts searches through the topology to find a node of kind mysocketio.
+// if that is found, the bindmounts are searched for ".mysocketio_token" and the path is being converted into an
+// absolute path and returned.
+// If the
+func deduceMySocketIoTokenFileFromBindMounts(nodes map[string]nodes.Node, configPath string) (string, error) {
+	for _, node := range nodes {
+		// search for mysocketio kind in topology config
+		if node.Config().Kind == "mysocketio" {
```

Comment:
lets maitain the happy path here and `continue` if kind is not mysocketio

Reply:
done